src/colorizer/utils/url_utils.ts

```diff
@@ -50,6 +51,123 @@ export function fetchWithTimeout(
   return fetchPromise;
 }
 
+/**
+ * Serializes the threshold into a string that can be used as a URL parameter.
+ *
+ * @param threshold FeatureThreshold to serialize.
+ * @returns A string representing the threshold.
+ * - For numeric features, the threshold is serialized as `featureName:unit:min:max`.
+ * - For categorical features, the threshold is serialized as `featureName:unit:selected_hex`,
+ * where `selected_hex` is the hex form of a binary number representing what categories are selected.
+ *
+ * The i-th place of the binary number is `1` if the i-th category in the feature's category list is enabled.
+ *
+ * ex: If there are five categories and the first and third categories are enabled,
+ * then `threshold.enabledCategories=[true, false, true, false, false]`.
+ * The binary representation is `00101`, which is `0x05` in hex.
+ */
+function serializeThreshold(threshold: FeatureThreshold): string {
+  // featureName + units are encoded in case it contains special characters (":" or ",").
+  // TODO: remove once feature keys are implemented.
+  const featureName = encodeURIComponent(threshold.featureName);
+  const featureUnit = encodeURIComponent(threshold.units);
+
+  // TODO: Are there better characters I can be using here? ":" and "," take up
+  // more space in the URL. -> once features are converted to use keys, use "-" as a separator here? "|"?
```

Comment:
agree that using keys with restricted character ranges is the way to go. but it will be a breaking change for compatibility with old existing urls!